internal/juju/resources_test.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package juju
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCharmResource_String(t *testing.T) {
+	cr := CharmResource{RevisionNumber: "5", OCIImageURL: "oci-url"}
+	assert.Equal(t, "5", cr.String(), "String() should return RevisionNumber if set")
+
+	cr = CharmResource{RevisionNumber: "", OCIImageURL: "oci-url"}
+	assert.Equal(t, "oci-url", cr.String(), "String() should return OCIImageURL if RevisionNumber is empty")
+}
+
+func TestCharmResource_Equal(t *testing.T) {
```

Comment:
Feels like we're testing the language a bit here (based on my other comment of just ==)